### DISCUSSION TITLE
How do you render a scene with a given resolution that differs from the original resolution in the scene XML file?

### DISCUSSION THREAD

**Discussion by ciy405x** [mitsuba-renderer/mitsuba3#763] [Q&A] [answered]
How do you render a scene with a given resolution that differs from the original resolution in the scene XML file?

I want to render several scenes with the resolution half of the original resolution in their XML file.
For instance, there are `a.xml`, `b.xml`, and `c.xml` files, and they have their distinct sensor descriptions in the XML file.
I want to do something like the below:
```python
for file in files:
    scene = mi.load_file(file)
    sensor = scene.sensors()[0]
    w, h = get_the_original_resolution_of_scene(sensor, ...)
    w /= 2
    h /= 2
    new_sensor = make_new_sensor_with_half_resolution(w, h, sensor, ...)
    output = mi.render(scene, sensor=new_sensor)
```

**Comment by njroussel** [comment, marked as answer]
Hi @ciy405x 

The `Film` exposes a `size` parameter through `mi.traverse()`. Take a look at the [`hdrfilm`](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_films.html#high-dynamic-range-film-hdrfilm) documentation for example. You should be able to modify this directly, without having to create a new sensor.

**Reply by ciy405x** [reply]
Thx for your hint!

```python
sensor_params = mi.traverse(scene.sensors()[0])
sensor_params["film.size"] //= 2
sensor_params.update()
```
